lib/opam_solve.ml

```diff
@@ -175,6 +178,22 @@ module Opam_monorepo_context (Base_context : BASE_CONTEXT) :
     in
     regular_versions @ avoid_versions
 
+  let prefer_version ~version versions =
+    let rec aux acc l =
+      match l with
+      | [] -> versions
+      | (v, opam) :: tl when OpamPackage.Version.equal v version ->
```

Comment:
```suggestion
      | ((version, _) as candidate) :: tl when OpamPackage.Version.equal version preferred_version ->
        candidate :: List.rev_append acc tl
```
feels more readable to me because it describes what the tuple is and comparing  `preferred_version` with `version` feels more understandable than comparing `version` with `v`.